File: 3315-construct-the-minimum-bitwise-array-ii/3315-construct-the-minimum-bitwise-array-ii.py
class Solution:
    def minBitwiseArray (self, nums : List [int]) -> List[int]:
        answer = [0] * len (nums)
        for i in range (len (nums)):
            k, n, ans = nums [i], 1, -1 
            while (k & n) != 0:
                ans = k - n
                n <<= 1
            answer [i] = ans
        return answer

File: 3315-construct-the-minimum-bitwise-array-ii/3315-construct-the-minimum-bitwise-array-ii.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* minBitwiseArray (int* nums, int numsSize, int* returnSize) {
    int* answer = (int*) malloc (sizeof (int) * numsSize);
    *returnSize = numsSize;
    for (int i = 0; i < numsSize; i++) {
        int k = nums [i], n = 1, ans = -1; 
        while ((k & n) != 0) {
            ans = k - n;
            n <<= 1;
        }
        answer [i] = ans;
    }
    return answer;
}